### PR TITLE
'Scipp'ification of the MDAnlysis parser 

### PR DESCRIPTION
**Pull request by Harry-Rich**
Implemented MDAnalysisparser with get_structure_coords_latt and get_indices. 

Things of note:

•Do we need to calculate volume within the parsers? - where do we then use it, I currently have not but can do if required, just an extra calc on the initial lattice. 
 
•Should  the fractional coordinates be dimensionless as discussed previously?
 
•Didn't implement the framework_indices stuff as you did not implement for the pymatgen parser, is this correct?
 
•Had to force the msd array to be float64 as MDAnalysis uses float32 by default
 
•I explicitly specified the function inputs for  _from_universe &  from_universe because something was breaking when i didn't will look into more tomorrow

### CODE REVIEW

File: kinisi/analyzer.py [hunk 1]
```diff
@@ -61,29 +63,60 @@ def _from_Xdatcar(cls,
...
             p.displacements = sc.concat([i.displacements for i in u], 'atom')
             return cls(p)
         elif dtype == 'consecutive':
             structures = _flatten_list([x.structures for x in trajectory])
             p = PymatgenParser(structures, specie, time_step, step_skip, dt, dimension, distance_unit, progress)
             return cls(p)
-        
+
+    @classmethod
+    def _from_Universe(cls,
```
Comment (arm61) [inline kinisi/analyzer.py:83]:
```suggestion
    def _from_universe(cls,
```

Python methods should be snake_case.

File: kinisi/diffusion_analyzer.py [hunk 2]
```diff
@@ -67,10 +67,38 @@ def from_Xdatcar(cls,
...
         """
-        p = super()._from_Xdatcar(trajectory, specie, time_step, step_skip, dtype, dt, dimension, distance_unit, progress)
+        p = super()._from_Xdatcar(trajectory, specie, time_step, step_skip, dtype, dt, dimension, distance_unit,
+                                  progress)
         p.msd = calculate_msd(p.trajectory, progress)
         return p
-    
+
+    @classmethod
+    def from_Universe(cls,
```
Comment (arm61) [inline kinisi/diffusion_analyzer.py:76]:
```suggestion
    def from_universe(cls,
```

File: kinisi/parser.py [hunk 3]
```diff
@@ -225,3 +225,129 @@ def get_indices(
...
+        structure: "MDAnalysis.universe.Universe",
+        specie: str,
+    ) -> Tuple[sc.Variable, sc.Variable]:
+        """
+        Determine framework and non-framework indices for an :py:mod:`MDAnalysis` compatible file.
+
+        :param structure: Initial structure.
+        :param specie: Specie to calculate diffusivity for as a String, e.g. :py:attr:`'Li'`.
+
+        :return: Tuple containing: indices for the atoms in the trajectory used in the calculation of the
```
Comment (arm61) [inline kinisi/parser.py:332]: Having a second colon in the line might break the automatic documentation generation. 
```suggestion
        :return: Tuple containing indices for the atoms in the trajectory used in the calculation of the
```

File: kinisi/parser.py [hunk 4]
```diff
@@ -225,3 +225,129 @@ def get_indices(
...
+        output of the positions in the trajectory. Similar to the :py:attr:`time_step`, this parameter must be
+        a :py:mod:`scipp` scalar. The units for this scalar should be dimensionless.
+    :param dt: Time intervals to calculate the displacements over. Optional, defaults to a :py:mod:`scipp` array
+        ranging from the smallest interval (i.e., time_step * step_skip) to the full simulation length, with 
+        a step size the same as the smallest interval.
+    :param dimension: Dimension/s to find the displacement along, this should be some subset of `'xyz'` indicating
+        the axes of interest. Optional, defaults to `'xyz'`.
+    :param distance_unit: The unit of distance used in the input structures. Optional, defaults to angstroms.
+    :param sub_sample_atoms: Subsample the atoms in the trajectory. Optional, defaults to 1.
+    :param sub_sample_traj: Subsample the trajectory. Optional, defaults to 1.
```
Comment (arm61) [inline kinisi/parser.py:251]: Perhaps we should aim to have parity between the different parsers in terms of features. Pymatgen doesn't have these options. Thoughts?

**Comment by arm61** [comment]
> Do we need to calculate volume within the parsers? - where do we then use it, I currently have not but can do if required, just an extra calc on the initial lattice.

I don't think we do. We can look again at it if necessary. 

> Should the fractional coordinates be dimensionless as discussed previously?

Yes. 

> Didn't implement the framework_indices stuff as you did not implement for the pymatgen parser, is this correct?

That's fine, that is a feature to come in the scipp version. 

> Had to force the msd array to be float64 as MDAnalysis uses float32 by default

Yes, MDA can be a bit annoying with that.

> I explicitly specified the function inputs for _from_universe & from_universe because something was breaking when i didn't will look into more tomorrow

I don't quite understand what you mean by this point.

File: kinisi/parser.py (hunk 4 above)
Comment (Harry-Rich) [reply kinisi/parser.py:251]: Agree r.e. parity, have updated

**Comment by Harry-Rich** [comment]
Fixed above points

**Pull request merged**